scheduler/docker/docker.go

```diff
@@ -146,73 +146,75 @@ func NewScheduler(client *dockerutil.Client) *Scheduler {
 	}
 }
 
-func (s *Scheduler) Run(ctx context.Context, app *twelvefactor.Manifest, p *twelvefactor.Process, in io.Reader, out io.Writer) error {
+func (s *Scheduler) Run(ctx context.Context, app *twelvefactor.Manifest, in io.Reader, out io.Writer) error {
 	attached := out != nil || in != nil
 
 	if !attached {
 		return errors.New("cannot run detached processes with Docker scheduler")
 	}
 
-	labels := twelvefactor.Labels(app, p)
-	labels[runLabel] = Attached
-
-	if err := s.docker.PullImage(ctx, docker.PullImageOptions{
-		Registry:     p.Image.Registry,
-		Repository:   p.Image.Repository,
-		Tag:          p.Image.Tag,
-		OutputStream: replaceNL(out),
-	}); err != nil {
-		return fmt.Errorf("error pulling image: %v", err)
-	}
+	for _, p := range app.Processes {
```

Comment:
This seems like a pretty big change - there will only ever be a single process, right?  Is this just short circuiting needing the `p` since that's already represented as an array in `app.Processes` and it only has a single value when `Run` is called?